docs/configuration/schema-config-reference.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "Schema Configuration"
+linkTitle: "Schema Configuration"
+weight: 1
+slug: schema-configuration
+---
+
+Cortex uses a NoSQL Store to store its index and optionally an Object store to store its chunks. Cortex has overtime evolved its schema to be more optimal and better fit the use-cases that arose. Currently there are 9 schemas that are used in production but we recommend running with `v9` schema when possible. You can move from one schema to another if a new schema fits your purpose better, but you still need to configure Cortex to make sure it can read the old data in the old schemas.
+
+You can configure the schemas using a YAML config file, that you can point to using the `-schema-config-file` flag. It has the following YAML spec:
+
+```yaml
+configs: []<period_config>
+```
+
+Where `period_config` is
+```
+from: <Date>
+store: <string> // The index client to use.
+object_store: <string> // The object client to use. If none is specified, `store` is used for storing chunks as well.
+schema: <string> // The schema version to use. Valid ones are v1, v2, v3,... v6, v9, v10, v11. Recommended for production: v9
+index: <periodic_table_config>
+chunks: <periodic_table_config>
+```
+
+Where `periodic_table_config` is
+```
+prefix: <string> // The prefix to use for the tables.
+period: <time.Duration> // We typically run Cortex with new tables every week to keep the index size low and to make retention easier. This sets the period at which new tables are created and used. Typically 168h (1week).
+tags: <map[string]string> // The tags that can be set on the dynamo table. 
+```
+
+Now an example of this file (also something recommended when starting out) is:
+```
+configs:
+  - from: "2020-03-01" // Or typically a week before the Cortex cluster was created.
+    schema: v9
+    index:
+      period: 168h
+      prefix: cortex_index_
+    chunks:                     // Chunks section is optional and required only if you're not using a separate object store.
+      period: 168h
+      prefix: cortex_chunks  
+    store: aws-dynamo/bigtable-hashed/cassandra/boltdb
+    object_store: <above options>/s3/gcs/azure/filesystem
+```
+
+An example of an advanced schema file with a lot of changes:
+```
+configs:
+  - from: "2018-08-23"
```

Comment:
For each config block I would add a `#` comment to explain what it does. For example, I'm adding a suggestion for the first block:

```suggestion
  # Starting from 2018-08-23 Cortex should store chunks and indexes
  # on Google BigTable using weekly periodic tables. The chunks table
  # names will be prefixed with "dev_chunks_", while index tables will be
  # prefixed with "dev_index_".
  - from: "2018-08-23"
```